src/native/corehost/apphost/static/CMakeLists.txt

```diff
@@ -151,23 +151,27 @@ else()
     include(${CLR_SRC_NATIVE_DIR}/libs/System.IO.Compression.Native/extra_libs.cmake)
     append_extra_compression_libs(NATIVE_LIBS)
 
-    # Additional requirements for System.Net.Security.Native
-    include(${CLR_SRC_NATIVE_DIR}/libs/System.Net.Security.Native/extra_libs.cmake)
-    append_extra_security_libs(NATIVE_LIBS)
+    if (NOT CLR_CMAKE_TARGET_TVOS) # no gssapi on tvOS, see https://developer.apple.com/documentation/gss
+        # Additional requirements for System.Net.Security.Native
+        include(${CLR_SRC_NATIVE_DIR}/libs/System.Net.Security.Native/extra_libs.cmake)
+        append_extra_security_libs(NATIVE_LIBS)
+    endif()
 
     # Additional requirements for System.Native
     include(${CLR_SRC_NATIVE_DIR}/libs/System.Native/extra_libs.cmake)
     append_extra_system_libs(NATIVE_LIBS)
 
-    # Additional requirements for System.Security.Cryptography.Native.OpenSsl
-    include(${CLR_SRC_NATIVE_DIR}/libs/System.Security.Cryptography.Native/extra_libs.cmake)
-    append_extra_cryptography_libs(NATIVE_LIBS)
+    if(NOT CLR_CMAKE_TARGET_MACCATALYST AND NOT CLR_CMAKE_TARGET_IOS AND NOT CLR_CMAKE_TARGET_TVOS)
+        # Additional requirements for System.Security.Cryptography.Native.OpenSsl
```

Comment:
Do we also need a condition for OpenSsl here?

https://github.com/dotnet/runtime/blob/2144482325b34adf70ec0b812159f5574f8b604d/src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.targets#L70

Reply:
yeah we could, but StaticOpenSslLinking isn't true by default and it doesn't make sense on Apple mobile so we could also make it an error if it is set

Reply:
also we split out most of the MSBuild side changes into https://github.com/dotnet/runtime/pull/82086 so would prefer making the change there

Reply:
`Static*Linking` properties have many sharp edges. I would not bother with explicit error handling for when they are used incorrectly.